lxc/lxc-openvdc/lxc-openvdc.go

```diff
@@ -0,0 +1,216 @@
+	}
+
+	if errorLog.Len() > 0 {
+		ioutil.WriteFile(filepath.Join(errorLogPath, "lxc-openvdc-ERROR.log"), errorLog.Bytes(), 0644)
+	}
+}
+
+func SetupContainerDir() error {
+	err := os.MkdirAll(rootfsPath, os.ModePerm)
+	if err != nil {
+		errors.Wrapf(err, "Failed creating container folder.")
+	}
+
+	if rootfsPath != "" {
+		err = DecompressXz("rootfs.tar.xz", rootfsPath)
+		if err != nil {
+			return err
+		}
+	} else {
+		return errors.New("RootfsPath not set.")
+	}
+
+	return nil
+}
+
+func PrepareCache() error {
+
+	if Exists(cacheFolderPath) == false {
+		err := CreateCacheFolder(cacheFolderPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	if Exists(filepath.Join(cacheFolderPath, "meta.tar.xz")) == false {
+		err := GetFile("meta.tar.xz")
+		if err != nil {
+			errors.Wrapf(err, "Failed downloading file.")
+		}
+		err = DecompressXz("meta.tar.xz", cacheFolderPath)
+		if err != nil {
+			errors.Wrapf(err, "Failed decompressing file.")
+		}
+	}
+
+	if Exists(filepath.Join(cacheFolderPath, "rootfs.tar.xz")) == false {
+		err := GetFile("rootfs.tar.xz")
+		if err != nil {
+			errors.Wrapf(err, "Failed downloading file.")
+		}
+	}
+
+	return nil
+}
+
+func GenerateConfig() error {
+	lxcCfgPath := filepath.Join(lxcPath, "config")
+	cfgPath := filepath.Join(containerPath, "config")
+	cacheCfgPath := filepath.Join(cacheFolderPath, "config")
+
+	f, err := ioutil.ReadFile(cacheCfgPath)
+	if err != nil {
+		errors.Wrapf(err, "Failed reading file: %s.", cacheCfgPath)
+	}
+
+	s := "\n# Distribution configuration\n" + string(f[:])
+	s = strings.Replace(s, "LXC_TEMPLATE_CONFIG", lxcCfgPath, -1)
+
+	s += fmt.Sprintf(
+`
+# Container specific configuration
+lxc.rootfs = %s
+lxc.utsname = %s
+ 
+# Network configuration
+lxc.network.type = veth
+lxc.network.flags = up
+lxc.network.link = virbr0
+`, rootfsPath, containerName)
+
+	b := []byte(s)
+
+	err = ioutil.WriteFile(cfgPath, b, 0644)
+	if err != nil {
+		errors.Wrapf(err, "Failed writing to file: %s.", cfgPath)
+	}
+
+	return nil
+}
+
+func GetFile(fileName string) error {
+
+	filePath := filepath.Join(cacheFolderPath, fileName)
+
+	res, err := http.Get("http://" + imgPath + "/" + fileName)
+	if err != nil {
+		return errors.Wrapf(err, "Failed Http.Get for file: %s", fileName)
+	}
+
+	defer res.Body.Close()
+
+	fileContent, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return errors.Wrapf(err, "Failed reading http response for file: %s", fileName)
+
+	}
+
+	f, err := os.Create(filePath)
+
+	if err != nil {
+		return errors.Wrapf(err, "Failed creating file: %s", fileName)
+	}
+	defer f.Close()
+
+	_, err = f.Write(fileContent)
+	if err != nil {
+		return errors.Wrapf(err, "Failed writing to file: %s", fileName)
+	}
+
+	return nil
+}
+
+func Exists(path string) bool {
+	if _, err := os.Stat(path); err != nil {
+		return false
+	}
+	return true
+}
+
+func CreateCacheFolder(folderPath string) error {
+	err := os.MkdirAll(folderPath, os.ModePerm)
+	if err != nil {
```

Comment:
It can simply write:

```
return errors.Wrapf(err, "Failed to create cache folder: %s", folderPath)
````

nil check is done inside of ``pkg/errors``.